### PR TITLE
#4126 fixed plain color layers, changed layer index to skip http layer

### LINKED ISSUE #4126: Missing support for plain background/background url layer at named map
@Kartones new definition:
Documentation for plain bg/image bg layers was missing at [windshaft](https://github.com/CartoDB/Windshaft/blob/0.44.1/doc/MapConfig-1.3.0.md), so was wrongly implemented as just an `http` layer with empty urlTemplate.
They use a different format and layer type instead of `http` (`plain` with a `color` property, and other field for background url). Implement them properly.

Old definition:
<hr>
for example, this: https://javi.cartodb.com/api/v1/map/static/named/tpl_7a99951e_aba7_11e3_b843_0e10bcd91c2b/300/300.png

steps:
- create a visualization with plain color baselayer
- add a cartodb layer
- request the static map

I guess it's because backend is not dealing properly with plain baselayers (take into account we have WMS layers too which can be translated to http ones)

### PR DESCRIPTION
**Pull request by Kartones**
Fixes #4126

### CODE REVIEW

**Comment by Kartones** [comment]
@javisantana @rochoa take a look please. All my tests have been ok with normal basemaps, plain colors, and BG ones (just skips the layer and renders others)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4444/))

**Comment by javisantana** [comment]
hard to read diff, more or less I understand your changes and look good

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/4445/))

File: services/named-maps-api-wrapper/lib/named-maps-wrapper/named_map.rb [hunk 1]
```diff
@@ -154,11 +154,15 @@ def self.get_template_data( visualization, parent )
         unless layer_group.nil?
           layer_group[:options][:layer_definition][:layers].each { |layer|
             layer_type = layer[:type].downcase
+
+            # INFO: Bypass image-bg layers for now
+            if layer_type == 'background' && (layer[:options]['image'].length > 0)
```
Comment (rochoa) [inline services/named-maps-api-wrapper/lib/named-maps-wrapper/named_map.rb]: Does this affect the layer indexes?

Comment (Kartones) [inline services/named-maps-api-wrapper/lib/named-maps-wrapper/named_map.rb]: no, diff doesn't shows properly, but layer index is not touched for any bg layer (as discussed), so skipping the bg with image doesn't either.

**Pull request merged**